packages/integrations/vue/src/index.ts

```diff
@@ -42,18 +49,39 @@ function virtualAppEntrypoint(options?: Options) {
 				return resolvedVirtualModuleId;
 			}
 		},
-		load(id: string) {
+		async load(id: string) {
+			const noop = `export const setup = () => {}`;
 			if (id === resolvedVirtualModuleId) {
-				if (options?.appEntrypoint) {
-					return `export { default as setup } from "${options.appEntrypoint}";`;
+				if (options.appEntrypoint) {
+					try {
+						let resolved;
+						if (options.appEntrypoint.startsWith('.')) {
+							resolved = await this.resolve(fileURLToPath(new URL(options.appEntrypoint, options.root)));
+						} else {
+							resolved = await this.resolve(options.appEntrypoint, fileURLToPath(options.root));
+						}
+						if (!resolved) {
+							throw new Error();
+						}
+						const loaded = await this.load(resolved);
+						if (!loaded.hasDefaultExport) {
```

Comment:
Rollup already has an API for loading a file, and the returned `ModuleInfo` has a handy `hasDefaultExport` property. This avoids us trying to load and parse the file ourselves, again avoiding a bunch of potential edge cases.